js/__internal/grids/grid_core/data_controller/module.ts

```diff
@@ -348,24 +348,27 @@ export class DataController extends ControllerWithDataMixin {
   }
 
   combinedFilter(filter, returnDataField?) {
```

Comment:
Let's refactor this function a little and make it shorter

Something like this:
```suggestion
combinedFilter(filter, returnDataField?) {
    if (!this._dataSource) {
      return filter;
    }

    // Can we use "??" here?
    let resultFilter = filter ?? this._dataSource.filter();
    const resultReturnDataField = returnDataField ?? this._dataSource.remoteOperations().filtering;
    const isColumnsTypesDefined = this._columnsController.isDataSourceApplied()
      || this._columnsController.isAllDataTypesDefined();

    if (isColumnsTypesDefined) {
      const additionalFilter = this._calculateAdditionalFilter();
      resultFilter = additionalFilter
        ? gridCoreUtils.combineFilters([additionalFilter, resultFilter])
        : resultFilter;
    }

    return this._columnsController.updateFilter(
      resultFilter,
      resultReturnDataField,
    );
  }
```